src/main/java/com/dataloom/datastore/edm/controllers/EdmController.java

```diff
@@ -196,6 +201,59 @@ public Schema getSchemaContents(
             @PathVariable( NAME ) String name ) {
         return schemaManager.getSchema( namespace, name );
     }
+    
+    private static void setDownloadContentType( HttpServletResponse response, FileType fileType ) {
+        if ( fileType == FileType.json ) {
+            response.setContentType( MediaType.APPLICATION_JSON_VALUE );
+        } else {
+            response.setContentType( CustomMediaType.TEXT_YAML_VALUE );
+        }
+    }
+
+    private static void setContentDisposition(
+            HttpServletResponse response,
+            String fileName,
+            FileType fileType ) {
+        if ( fileType == FileType.yaml || fileType == FileType.json ) {
+            response.setHeader( "Content-Disposition",
+                    "attachment; filename=" + fileName + "." + fileType.toString() );
+        }
+    }
+    
+    @RequestMapping(
+            path = SCHEMA_PATH + NAMESPACE_PATH + NAME_PATH,
+            method = RequestMethod.GET,
+            produces = { MediaType.APPLICATION_JSON_VALUE, CustomMediaType.TEXT_YAML_VALUE } )
+    public String getSchemaContentsFormatted(
```

Comment:
Return Schema and just register the YamlMapper as handling the custom media. It will automatically handle selecting the write mapper when serializing the schemas